libcasr/src/js.rs

```diff
@@ -0,0 +1,390 @@
+//! JS module implements `ParseStacktrace` and `Exception` traits for JS reports.
+use crate::error::{Error, Result};
+use crate::exception::Exception;
+use crate::execution_class::ExecutionClass;
+use crate::stacktrace::{ParseStacktrace, Stacktrace, StacktraceEntry};
+
+use regex::Regex;
+
+/// Structure provides an interface for parsing JS exception message.
+pub struct JSException;
+
+impl Exception for JSException {
+    fn parse_exception(stderr: &str) -> Option<ExecutionClass> {
+        // TODO: discuss the format of the 1st string
+        let rexception = Regex::new(r"^(.*Error):(?:\s+(.*))?$").unwrap();
+        let Some(captures) = rexception.captures(stderr) else {
+            return None;
+        };
+        let error_type = if let Some(error_type) = captures.get(1) {
+            error_type.as_str()
+        } else {
+            captures.get(2).unwrap().as_str()
+        };
+        let message = captures.get(2).unwrap().as_str();
+        Some(ExecutionClass::new((
+            "NOT_EXPLOITABLE",
+            error_type,
+            message,
+            "",
+        )))
+    }
+}
+
+/// Structure provides an interface for processing the stack trace.
+pub struct JSStacktrace;
+
+impl ParseStacktrace for JSStacktrace {
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>> {
+        // Get stack trace from JS report.
+        // TODO: discuss the format of the 1st string
+        let re = Regex::new(r"(?m)^(?:.*Error)(?:.|\n)*?((?:\n(?:\s|\t)*at .*)+)").unwrap();
```

Comment:
Isn't \t included in \s?

Reply:
Indeed, it's included  
Fixed also in java.rs  
Done